File: Lab-08-solution/petclinic/src/main/java/com/petclinic/core/Owner.java
package com.petclinic.core;

import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;

@Entity
public class Owner {
    @Id @GeneratedValue
    private int id;
    private String firstName;
    private String lastName;
    private double accountStatement;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "owner_id")
    private List<Pet> pets = new ArrayList<>();

    public Owner() {
    }

    public Owner(int id, String firstName, String lastName, double accountStatement) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.accountStatement = accountStatement;
    }

    public int getId() {
        return id;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public double getAccountStatement() {
        return accountStatement;
    }

    public void setAccountStatement(double accountStatement) {
        this.accountStatement = accountStatement;
    }

    public List<Pet> getPets() {
        return pets;
    }

    public void setPets(List<Pet> pets) {
        this.pets = pets;
    }
}

File: Lab-kotlin/src/main/kotlin/com/petclinic/core/Owner.kt
package com.petclinic.core

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
class Owner(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    var firstName: String,
    var lastName: String,
    var accountStatement: BigDecimal,

    @OneToMany(cascade = [CascadeType.PERSIST])
    @JoinColumn(name="owner_id")
    var pets: List<Pet>?
)
